lib/bolt/cli.rb

```diff
@@ -559,6 +563,50 @@ def initialize_project
       ok ? 0 : 1
     end
 
+    def migrate_project
+      if inventory.version == 2
+        ok = true
+      else
+        inventory_file = config.inventoryfile || config.boltdir.inventory_file
+
+        begin
+          Bolt::Util.file_stat(inventory_file)
+        rescue Errno::ENOENT
+          raise Bolt::FileError.new("The inventory file '#{inventory_file}' does not exist", inventory_file)
+        end
+
+        inv = YAML.safe_load(File.open(inventory_file))
+        migrate_group(inv)
+
+        ok = File.write(inventory_file, { 'version' => 2 }.merge(inv).to_yaml)
+      end
+
+      result = if ok
+                 "Successfully migrated Bolt project to latest version"
+               else
+                 "Could not migrate Bolt project to latest version"
+               end
+      outputter.print_message result
+
+      ok ? 0 : 1
+    end
+
+    # Walks an inventory hash and replaces all 'nodes' keys with 'targets' keys
+    # and all 'name' keys nested in a 'targets' hash with 'uri' keys
+    def migrate_group(group)
+      if group.key?('nodes')
+        targets = group['nodes'].map do |target|
+          target['uri'] = target.delete('name') if target.is_a?(Hash)
+          target
+        end
+        group.delete('nodes')
+        group['targets'] = targets
+      end
+      (group['groups'] || []).map do |subgroup|
+        migrate_group(subgroup)
```

Comment:
Maybe `map` is confusing given the data structure is modified in place. We could use each, and explicitly return `nil` to avoid confusion on expected return value. We should also update the comment about mutating the data. 
```ruby
    def migrate_group(group)
      if group.key?('nodes')
        targets = group['nodes'].map do |target|
          target['uri'] = target.delete('name') if target.is_a?(Hash)
          target
        end
        group.delete('nodes')
        group['targets'] = targets
      end
      (group['groups'] || []).each { |subgroup| migrate_group(subgroup)} 
      nil
    end
```

Reply:
Updated.